tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -599,6 +602,7 @@ public void onSubscribe(Subscription s) {
       env.debug(String.format("%s::onSubscribe(%s)", this, s));
       if (!subscription.isCompleted()) {
         subscription.complete(s);
```

Comment:
There is a [completeImmediately()](https://github.com/reactive-streams/reactive-streams-jvm/blob/master/tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java#L910) method on the promise.

Reply:
Btw. the plain `complete()` behavior is odd. I'd think once called, `isCompleted()` should report true but the implementation queues the object.

Reply:
I agree - but I'm not familiar enough with the TCK to change it. Though I can understand why the event is queued, so that the order of events can be verified and duplicates can be detected etc.

Reply:
I'll investigate this

Reply:
Thanks for noticing, I think we can improve the impl there. I'll work on it today after done with work stuff.
I'll PR into this PR then :)

Reply:
I reimplemented the Promise which makes this workwithout the additional atomic.
Please see here: https://github.com/jroper/reactive-streams-jvm/pull/1